core/src/main/java/org/sonarsource/sonarlint/core/telemetry/TelemetryLocalStorage.java

```diff
@@ -43,12 +45,22 @@
   private int showHotspotRequestsCount;
   private int taintVulnerabilitiesInvestigatedLocallyCount;
   private int taintVulnerabilitiesInvestigatedRemotelyCount;
+  private final Collection<String> raisedIssuesRules;
 
   TelemetryLocalStorage() {
     enabled = true;
     installTime = OffsetDateTime.now();
     analyzers = new LinkedHashMap<>();
     notificationsCountersByEventType = new LinkedHashMap<>();
+    raisedIssuesRules = new HashSet<>();
+  }
+
+  public Collection<String> getRaisedIssuesRules() {
+    return raisedIssuesRules;
+  }
+
+  public void addReportedRule(String reportedRule) {
+    this.raisedIssuesRules.add(reportedRule);
```

Comment:
you should clear the set in the `clearAfterPing` method (and add a test) because if not, rules will continue to accumulate days after days

Reply:
Done.